consensus/state.go

```diff
@@ -630,6 +630,13 @@ func (ms *MidState) fileContractElement(ts V1TransactionSupplement, id types.Fil
 	return ts.revision(id)
 }
 
+func (ms *MidState) storageProof(ts V1TransactionSupplement, id types.FileContractID) (V1StorageProofSupplement, bool) {
+	if i, ok := ms.created[id]; ok && ms.fces[i].FileContract.WindowStart == ms.base.childHeight() {
```

Comment:
Should this return immediately if `ms.fces[i].FileContract.WindowStart == ms.base.childHeight()`?

Reply:
not sure I follow, but we could write it this way:
```go
if ms.fces[i].FileContract.WindowStart == ms.base.childHeight() {
    if 
    return V1StorageProofSupplement{FileContract: ms.fces[i], WindowID: ms.base.Index.ID}, 
}
```
basically, there are two possibilities:
- The contract was created in an earlier block. In that case, it will be present in the transaction supplement, so we find it via `ts.storageProof`.
- The contract was created in this block. In that case, the WindowStart *must* be the next block; otherwise, we can't provide the WindowID (since it would be the ID of a future block).

Reply:
That was basically my thought. As written, if `ok` is true and `WindowStart == ms.base.childHeight()` is false it will call `ts.storageProof` and loop through the transaction supplement which shouldn't include the contract ID since it was just created. It's unlikely to save many cycles, but the extra loop seemed unnecessary.

Reply:
I ended up always returning `true`, even if the window start is different. This is a bit odd, but it means that for early storage proofs, we return a more informative error rather than "nonexistent contract".

Reply:
Always returning `true` doesn't seem right. If there was ever a bug that managed to call `Apply` without `Validate`, it would silently finalize the contract instead of panicking. I agree that the error is unhelpful though. Maybe `storageProof` could return an error instead?